src/CitizenProject/CitizenProjectListener.php

```diff
@@ -0,0 +1,54 @@
+<?php
+
+namespace AppBundle\CitizenProject;
+
+use AppBundle\Entity\CitizenProject;
+use Doctrine\ORM\Event\LifecycleEventArgs;
+use League\Flysystem\Filesystem;
+use League\Glide\Server;
+use Symfony\Component\HttpFoundation\File\UploadedFile;
+
+class CitizenProjectListener
+{
+    /**
+     * @var Filesystem
+     */
+    private $storage;
+
+    /**
+     * @var Server
+     */
+    private $glide;
+
+    public function __construct(Filesystem $storage, Server $glide)
+    {
+        $this->storage = $storage;
+        $this->glide = $glide;
+    }
+
+    public function preUpdate(CitizenProject $citizenProject, LifecycleEventArgs $event)
+    {
+        $this->addImage($citizenProject);
+    }
+
+    public function prePersist(CitizenProject $citizenProject, LifecycleEventArgs $event)
+    {
+        $this->addImage($citizenProject);
+    }
+
+    private function addImage(CitizenProject $citizenProject): void
+    {
+        // Save citizen project image to cloud storage
+        if ($citizenProject->getImage() instanceof UploadedFile) {
```

Comment:
What if the person update the project without editing the image ? The image would be uploaded again ?

Reply:
No. `$citizenProject->getImage()` returns an object `UploadedFile` only in case where file is uploaded.